lang/cs/Org.Apache.REEF.Network/Elastic/Operators/Logical/ElasticOperator.cs

```diff
@@ -0,0 +1,552 @@
+        /// <returns>The configuration for the task with added operator information</returns>
+        internal void GetTaskConfiguration(ref IList<string> serializedOperatorsConfs, int taskId)
+        {
+            if (_operatorFinalized && _operatorStateFinalized)
+            {
+                GetOperatorConfiguration(ref serializedOperatorsConfs, taskId);
+
+                _next?.GetTaskConfiguration(ref serializedOperatorsConfs, taskId);
+            }
+            else
+            {
+                throw new IllegalStateException("Operator needs to be finalized before getting tasks configuration.");
+            }
+        }
+
+        /// <summary>
+        /// Whether this operator is ready to be scheduled by the task set manager.
+        /// </summary>
+        /// <returns>True if the operator is ready to be scheduled</returns>
+        internal bool CanBeScheduled()
+        {
+            return _topology.CanBeScheduled() && (_next?.CanBeScheduled() ?? true);
+        }
+
+        /// <summary>
+        /// Utility method gathering the set of master task ids of the operators in the current pipeline.
+        /// </summary>
+        /// <param name="masterTasks">The id of the master tasks of the current and successive operators</param>
+        internal virtual void GatherMasterIds(ref HashSet<string> masterTasks)
+        {
+            if (!_operatorFinalized)
+            {
+                throw new IllegalStateException("Operator need to be build before gathering information.");
+            }
+
+            masterTasks.Add(Utils.BuildTaskId(Stage.StageName, MasterId));
+
+            _next?.GatherMasterIds(ref masterTasks);
+        }
+
+        /// <summary>
+        /// Log the final statistics of the operator.
+        /// This is called when the pipeline execution is completed.
+        /// </summary>
+        internal virtual string LogFinalStatistics()
+        {
+            return LogInternalStatistics() + _next?.LogFinalStatistics();
+        }
+
+        /// <summary>
+        /// Appends the message type to the configuration.
+        /// </summary>
+        /// <param name="operatorType">The type of the messages the operator is configured to accept</param>
+        /// <returns>The conf builder with added the message type</returns>
+        protected IConfiguration SetMessageType(Type operatorType)
+        {
+            if (operatorType.IsGenericType)
+            {
+                var genericTypes = operatorType.GenericTypeArguments;
+                var msgType = genericTypes[0];
+
+                return TangFactory.GetTang().NewConfigurationBuilder()
+                    .BindStringNamedParam<OperatorParameters.MessageType>(msgType.AssemblyQualifiedName)
+                    .Build();
+            }
+            else
+            {
+                throw new IllegalStateException("Expecting a generic type for the message.");
```

Comment:
I added this method because otherwise I had to repeat the same code for all different operators. Let me put your implementation (without the `TOperator`).